nebula_decoders/include/nebula_decoders/nebula_decoders_velodyne/decoders/vlp_16.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include "nebula_decoders/nebula_decoders_velodyne/decoders/velodyne_sensor.hpp"
+
+namespace nebula
+{
+namespace drivers
+{
+
+class VLP16 : public VelodyneSensor
+{
+public:
+  uint16_t getAzimuthCorrected(
+    uint16_t azimuth, float azimuth_diff, int firing_sequence, int firing_order)
+  {
+    float azimuth_corrected =
+      azimuth + (azimuth_diff *
+                 ((firing_order * VLP16_DSR_TOFFSET) + (firing_sequence * VLP16_FIRING_TOFFSET)) /
+                 VLP16_BLOCK_DURATION);
+
+    return static_cast<uint16_t>(round(azimuth_corrected)) % 36000;
+  }
+
+  constexpr static int num_maintenance_periods = 0;
+
+  constexpr static int num_simultaneous_firings = 1;
+
+  constexpr static double firing_sequences_per_block = 2.0;
+
+  constexpr static int channels_per_firing_sequence = 16;
+
+  constexpr static float distance_resolution_m = 0.002f;
+
+  constexpr static double full_firing_cycle_s = 55.296 * 1e-6;
+
+  constexpr static double single_firing_s = 2.304 * 1e-6;
+
+  constexpr static double offset_packet_time = 0;
```

Comment:
These constants should ideally be implemented as arguments to the parent class (`VelodyneSensor`) constructor or template.
As it is, there is no guarantee that each child class defines all the constants accessed by the decoder (a new developer implementing a new sensor might not realize they have to define those constants for the decoder to work).
If these constants need to be passed to a constructor, there is no way that anyone forgets to put them here.